### PR TITLE
Rewrite of JS version to pass tests and work faster

### PR DESCRIPTION
**Pull request by shedar**
Two major changes in this version:
- it passes tests
- it works much faster on 5M lines

Here is a description what was changed and how it affects performance https://olg.io/blog/2016/nodejs-wordcount/  in case someone is interested in details.

It will still fail on full data set due to nodeJS heap limits (1.5Gb by default, can be increased to 4Gb with command line options, but it's not enough). There is a way to workaround limitations by using of ArrayBuffer. But it requires a lot of effort to model required data structures manually in memory chunks. Maybe I'll find time to implement a version for full data set later.

### CODE REVIEW

File: javascript/wordcount.js [hunk 1]
```diff
@@ -1,51 +1,44 @@
...
-      return 1;
-
-    return -1;
+}).on('end', function () {
+  if (lastChunk) {
+    counts.set(lastChunk, (counts.get(lastChunk) || 0) + 1);
   }
-
-  TimSort.sort(wordList, compare);
+  var res = [...counts.keys()];
```
Comment (juditacs) [inline javascript/wordcount.js]: I get a syntax error at this line.

~~~
nodejs javascript/wordcount.js 
/root/wordcount/javascript/wordcount.js:26
  var res = [...counts.keys()];
             ^^^

SyntaxError: Unexpected token ...
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3
~~~

Comment (shssoichiro) [inline javascript/wordcount.js]: Which version of nodejs is this running on? Older versions don't have support for ES6 syntax and will error on this.

**Comment by shedar** [comment]
Spread operator is supported in 5.x.x versions of node. It's a stable version, so I used it even though it's not an LTS.

File: javascript/wordcount.js (hunk 1 above)
Comment (juditacs) [inline javascript/wordcount.js]: 4.4.1. I'll try to update it.

**Comment by juditacs** [comment]
Can you update the Dockerfile to install a newer version?

**Comment by shedar** [comment]
Sure. I'll update it either later today or tomorrow.

**Comment by shedar** [comment]
Does github miss notifications  about new commit or this PR still has issues?

**Comment by juditacs** [comment]
Yes, github does **not** send notifications about new commits. I only get notifications about new comments.

**Comment by juditacs** [comment]
I'm merging this, but it's not yet tested.

**Pull request merged**